File: config/convertToDeltaBatch.scala
import org.apache.spark.sql.{SaveMode, SparkSession}
//this script converts parquet files in the input dir to delta files in the outpu dir. delta tables have the same name as the parquet files 

object ConvertParquetToDeltaScript {
  def main(args: Array[String]): Unit = {
    val inputPath = args(0)
    val outputPath = args(1)

    convertParquetToDelta(inputPath, outputPath)
  }



  def convertParquetToDelta(inputPath: String, outputPath: String): Unit = {
    val spark = SparkSession.builder()
      .appName("ConvertParquetToDelta")
      .getOrCreate()

    val fileList = spark.read.format("parquet").load(inputPath).inputFiles

    fileList.foreach { file =>
      val df = spark.read.format("parquet").load(file)
      val fileName = file.split("/").last
      val deltaPath = s"$outputPath/$fileName"
      df.write.format("delta").mode(SaveMode.Append).save(deltaPath)
    }

    //spark.stop()
  }
}

File: config/convertToDeltaBatch.py
from pyspark.sql import SparkSession
from pyspark.sql import SaveMode

def convert_parquet_to_delta(input_path, output_path):
    spark = SparkSession.builder \
        .appName("ConvertParquetToDelta") \
        .getOrCreate()

    file_list = spark.read.format("parquet").load(input_path).inputFiles()

    for file in file_list:
        df = spark.read.format("parquet").load(file)
        file_name = file.split("/")[-1]
        delta_path = f"{output_path}/{file_name}"
        df.write.format("delta").mode(SaveMode.Append).save(delta_path)

input_path = "/path/to/input/dir"
output_path = "/path/to/output/dir"

convert_parquet_to_delta(input_path, output_path)
